### PR TITLE
fix(general): fix-up `*Fault` with lock

### PR DESCRIPTION
**Pull request by aaron-kasten**
Fixes https://github.com/kopia/kopia/issues/2848

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -70,6 +70,8 @@ linters-settings:
 linters:
   enable-all: true
   disable:
+    - wsl
+    - staticcheck
```
Comment (Shrekster) [inline .golangci.yml:74]: enable these ?

File: internal/fault/fault.go [hunk 2]
```diff
@@ -1,14 +1,18 @@
...
+	"time"
+)
 
-// Fault describes the behavior of a single fault.
+// Fault describes the behavior of a singe fault.
 type Fault struct {
 	repeatCount int           // how many times to repeat this fault
 	sleep       time.Duration // sleep before returning
 	callback    func()
 	errCallback func() error
```
Comment (Shrekster) [inline internal/fault/fault.go:14]: lets add checklocks

```suggestion
	// how many times to repeat this fault
	repeatCount int // +checklocks:mu
	// sleep before returning
	sleep       time.Duration // +checklocks:mu
	callback    func() // +checklocks:mu
	errCallback func() error // +checklocks:mu
```

**Review by Shrekster — APPROVED** [review]
Approved once the comments are resolved.

File: internal/fault/fault.go [hunk 3]
```diff
@@ -1,14 +1,20 @@
...
-	repeatCount int           // how many times to repeat this fault
-	sleep       time.Duration // sleep before returning
-	callback    func()
-	errCallback func() error
+	// how many times to repeat this fault
+	repeatCount int // +checklocks:mu
+	// sleep before returning
+	sleep       time.Duration // +checklocks:mu
+	callback    func() // +checklocks:mu
+	errCallback func() error // +checklocks:mu
```
[the remark this answers is not in the document]
Comment (Shrekster) [reply internal/fault/fault.go:16]: Ah, my bad. Yes let’s stay consistent.

File: internal/fault/fault.go [hunk 4]
```diff
@@ -18,30 +24,40 @@ func New() *Fault {
 
 // ErrorInstead causes the fault to return the provided error instead of calling the method.
 func (f *Fault) ErrorInstead(err error) *Fault {
+	f.mu.Lock()
 	f.errCallback = func() error { return err }
+	f.mu.Unlock()
```
Comment (julio-lopez) [inline internal/fault/fault.go:29]: fix(lint): wsl

```suggestion
	f.mu.Unlock()

```

File: internal/fault/fault.go [hunk 5]
```diff
@@ -18,30 +24,40 @@ func New() *Fault {
...
 	return f
 }
 
 // ErrorCallbackInstead invokes the provided function to return the error instead of calling the method.
 func (f *Fault) ErrorCallbackInstead(cb func() error) *Fault {
+	f.mu.Lock()
 	f.errCallback = cb
+	f.mu.Unlock()
```
Comment (julio-lopez) [inline internal/fault/fault.go:37]:
```suggestion
	f.mu.Unlock()

```

File: internal/fault/fault.go [hunk 6]
```diff
@@ -18,30 +24,40 @@ func New() *Fault {
...
 	return f
 }
 
 // Before invokes the provided function but does not return an error.
 func (f *Fault) Before(cb func()) *Fault {
+	f.mu.Lock()
 	f.callback = cb
+	f.mu.Unlock()
```
Comment (julio-lopez) [inline internal/fault/fault.go:45]:
```suggestion
	f.mu.Unlock()

```

File: internal/fault/fault.go [hunk 7]
```diff
@@ -18,30 +24,40 @@ func New() *Fault {
...
 	return f
 }
 
 // Repeat causes the fault to repeat N times.
 func (f *Fault) Repeat(n int) *Fault {
+	f.mu.Lock()
 	f.repeatCount = n
+	f.mu.Unlock()
```
Comment (julio-lopez) [inline internal/fault/fault.go:53]:
```suggestion
	f.mu.Unlock()

```

File: internal/fault/fault.go [hunk 8]
```diff
@@ -18,30 +24,40 @@ func New() *Fault {
...
 	return f
 }
 
 // SleepFor sleeps for the specified amount of time.
 func (f *Fault) SleepFor(d time.Duration) *Fault {
+	f.mu.Lock()
 	f.sleep = d
+	f.mu.Unlock()
```
Comment (julio-lopez) [inline internal/fault/fault.go:61]:
```suggestion
	f.mu.Unlock()

```

**Review by julio-lopez — APPROVED** [review]
🥇Thanks for doing this.

LGTM, feel free to merge once the linter and spelling issues are addressed.

File: internal/fault/fault.go [hunk 9]
```diff
@@ -1,14 +1,20 @@
...
-	sleep       time.Duration // sleep before returning
-	callback    func()
-	errCallback func() error
+	// how many times to repeat this fault
+	repeatCount int // +checklocks:mu
+	// sleep before returning
+	sleep       time.Duration // +checklocks:mu
+	callback    func() // +checklocks:mu
+	errCallback func() error // +checklocks:mu
+	mu          sync.Mutex
```
Comment (julio-lopez) [inline internal/fault/fault.go:17]:
```suggestion
```

**Pull request merged**